src/ensembl/src/content/app/entity-viewer/shared/rest/rest-adaptors/rest-transcript-adaptor.ts

```diff
@@ -140,29 +63,20 @@ const buildExon = (
   };
 };
 
-const buildCDS = (
-  transcript: TranscriptInResponse,
-  data: FeatureInResponse[]
-): CDS | null => {
-  const cdss = data.filter(
-    (feature) =>
-      feature.feature_type === 'cds' && feature.Parent === transcript.id
-  );
-
-  if (!cdss.length) {
-    return null;
-  }
-
-  const sortedCdss = sortBy(cdss, (cds) => cds.start);
-  const firstCds = sortedCdss[0];
-  const lastCds = sortedCdss[cdss.length - 1];
-
+const buildCDSFromLookup = (
+  translation: TranslationInResponse,
+  transcript: TranscriptInResponse
+): CDS => {
   return {
-    start: firstCds.start,
-    end: lastCds.end,
+    protein: {
+      id: translation.id,
+      length: translation.length
+    },
```

Comment:
The CDS that the graphql api is going to return to us is not going to have a `protein` field. Currently the closest shape that we got to in core data modelling looks something like this: https://gitlab.ebi.ac.uk/andrey/core-data-modelling/blob/master/splicing/hemE.json

I understand that what we are building now is very temporary, but probably better have a `protein` or `product` field on the transcript itself rather than on the CDS. Both options are wrong, but having the protein/product field on the transcript rather than on CDS is slightly less wrong.

Reply:
I've made changes to the `Transcript` structure to be more close to the CDM object.